test/test_expanded_weights.py

```diff
@@ -474,6 +475,38 @@ def forward(self, input):
         expected_grads = tuple(expected_grad for expected_grad in expected_grads if expected_grad is not None)
         assert [self.assertEqual(actual, 2 * expected) for (actual, expected) in zip(actual_grads, expected_grads)]
 
+    # getting the expected per sample grads for packed sequences is non-trivial. This just smoke checks that the
+    # outputs are the same and the sum of the per sample gradients is the same as the batched gradient
+    def _do_test_rnn_packed_sequence(self, module, input, args=None, kwargs=None):
+        args = args if args is not None else ()
+        kwargs = kwargs if kwargs is not None else {}
+
+        batch_size = max(tuple(input.batch_sizes)).item()
+
+        with freeze_rng_state():
+            # get per sample grads with ExpandedWeights context manager
+            actual_res = call_for_per_sample_grads(module,
+                                                   batch_size=batch_size,
+                                                   loss_reduction="sum")(input, *args, **kwargs).data.sum()
+            actual_res.backward(retain_graph=True)
```

Comment:
nit: don't need retain_graph since we don't backward through actual-res again, right? Unless I'm misreading

Reply:
Yeah great question I think I just was trying to make it work instead of actually looking into what's happening. Basically with packed sequences, this got weird since I was setting requires_grad on the tensor used to make the packed sequence. I wasn't able to unset the grad of that tensor since I only had access to the PackedSequence (which only has a tensor that's the output from a function, not the original tensor)